lib/ansible/modules/network/fortios/fortios_cmdb_set.py

```diff
@@ -0,0 +1,728 @@
+         wireless-controller wtp, wireless-controller wtp-group, \
+         wireless-controller wtp-profile]
+
+    mkey:
+        description:
+            - Master key of the resource for those endpoints \
+            representing a table
+
+    config_parameters:
+        description:
+            - Any of the additional parameters that are required by some \
+            endpoints to complete the request
+        required: true
+'''
+
+EXAMPLES = '''
+- hosts: localhost
+  vars:
+   host: "192.168.40.8"
+   username: "admin"
+   password: ""
+   vdom: "root"
+  tasks:
+  - name: Set static route on the fortigate
+    fortios_cmdb_set:
+     host:  "{{  host }}"
+     username: "{{  username}}"
+     password: "{{ password }}"
+     vdom:  "{{  vdom }}"
+     endpoint: "router static"
+     config_parameters:
+       seq-num: "8"
+       dst: "10.10.32.0 255.255.255.0"
+       device: "port2"
+       gateway: "192.168.40.252"
+  - name:   firewall policy
+    fortios_cmdb_set:
+     endpoint: "firewall policy"
+     host: "{{ host }}"
+     username: "{{ username }}"
+     password: "{{ password }}"
+     vdom:  "{{  vdom }}"
+     config_parameters:
+        policyid: "66"
+        name: "ansible"
+        json:
+          policyid: "66"
+          name: "ansible"
+          action: "accept"
+          srcintf: [ {"name": "port1"} ]
+          dstintf: [{"name":"port2"} ]
+          srcaddr: [{"name":"all"} ]
+          dstaddr: [{"name":"all"}]
+          schedule: "always"
+          service:  [{"name":"HTTPS"}]
+          logtraffic: "all"
+'''
+
+RETURN = '''
+results:
+  description: Data returned by the endpoint operation
+  returned: on sucess
+  type: string
+  sample: {
+            "apply-to": "admin-password ipsec-preshared-key",
+            "change-4-characters": "disable",
+            "expire-day": 90,
+            "expire-status": "enable",
+            "min-lower-case-letter": 2,
+            "min-non-alphanumeric": 1,
+            "min-number": 2,
+            "min-upper-case-letter": 2,
+            "minimum-length": 8,
+            "reuse-password": "disable",
+            "status": "enable
+          }
+
+status:
+  description: Indication of the operation's result
+  returned: always
+  type: string
+  sample: "success"
+
+version:
+  description: Version of the FortiGate
+  returned: always
+  type: string
+  sample: v5.6.2
+
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import logging
+
+try:
+    from fortiosapi import FortiOSAPI
+    HAS_FORTIOSAPI = True
+except ImportError:
+    HAS_FORTIOSAPI = False
+
+if HAS_FORTIOSAPI:
+    fos = FortiOSAPI()
+else:
+    raise ImportError("fortiosapi module is required")
+
+formatter = logging.Formatter('%(asctime)s %(name)-12s '
+                              '%(levelname)-8s %(message)s')
+logger = logging.getLogger('fortiosapi')
+hdlr = logging.FileHandler('/var/tmp/ansible-fortiosconfig.log')
```

Comment:
Will this have issues if two or more instances or ansible-playbook are running at the same time?

Reply:
Basically the messages from both instances will end up in the same file. This is intended to provide a log for the fortiosapi library which is called by this module. Normally it is not a problem to have messages mixed (messages are very simple) but if required I can add a random id/cookie to every message printed out.

Reply:
I'm concerned with this being the default behavior.  In general we should not be creating files on users machines that didn't ask for.  Maybe a flag to enable this, but it should be off by default.  I'm also questioning why this should be set to `DEBUG` also.  Maybe it should only be enabled with `ANSIBLE_DEBUG` is enabled.

Regardless, I think this should be changed or removed.

Reply:
Regarding previous fortios_config.py, we wanted to complement the functionality given by that module. To do that we prefer to rely on Fortinet's fortiosapi library which is intended to have more generic support for this and the rest of features that will come next. Fortiosapi has been designed with FortiGate's REST API in mind, to simplify user experience (endpoints in playbook are mapped to a rest call that the user is already familiar with). And given the large REST API of the FortiGate, the module can be easily maintained and updated.

Reply:
Agree on the default behavior it is a remains of the debugging developing. Will change that. Is there a better practice in Ansible to adapt to python logging we could follow ??

Reply:
Agree about the log. I'm removing it now. Will submit changes shortly.